software/py/vanilla_stats_parser.py

```diff
@@ -0,0 +1,448 @@
+      for y in range(self.manycore_dim_y):
+        for x in range(self.manycore_dim_x):
+          trace_idx -= 1
+          trace = self.traces[trace_idx]
+          for stat in stats_list:
+            self.tile_stat_dict[y][x][stat] = trace[stat]
+          for instr in instr_list:
+            self.tile_stat_dict[y][x][instr] = trace[instr]
+          for stall in stalls_list:
+            self.tile_stat_dict[y][x][stall] = trace[stall]
+          for miss in miss_list:
+            self.tile_stat_dict[y][x][miss] = trace[miss]
+
+
+  # Print execution timing for the entire manycore 
+  def print_manycore_stats_timing(self, stat_file):
+    # For total execution time, we only sum up the execution time of origin tile in tile groups
+    # The origin tile's relative coordinates is always 0,0 
+    stat_file.write("Timing Stats\n")
+    stat_file.write(stats_timing_header_format.format("tile group", "exec time", "share (%)"))
+    stat_file.write("{}".format(separating_line))
+
+    # Sum up execution time for all tile groups 
+    for i in range (0, self.num_tile_groups):
+      self.total_execution_time += (self.timing_end_list[i] - self.timing_start_list[i])
+
+    for i in range (0, self.num_tile_groups):
+      tg_time = self.timing_end_list[i] - self.timing_start_list[i]
+      stat_file.write(stats_timing_data_format.format(i,
+                                                      tg_time,
+                                                      (tg_time / self.total_execution_time * 100)))
+
+    stat_file.write(stats_timing_data_format.format("total",
+                                                    self.total_execution_time,
+                                                    (self.total_execution_time / self.total_execution_time * 100)))
+    stat_file.write("{}\n".format(separating_line))
+    return
+
+
+  # Print instruction stats for the entire manycore
+  def print_manycore_stats_instructions(self, stat_file):
+
+    stat_file.write("Instruction Stats\n")
+    stat_file.write(stats_instr_header_format.format("instruction", "count", " share (%)"))
+    stat_file.write("{}".format(separating_line))
+
+   
+    # Print instruction stats for manycore
+    for instr in instr_list:
+       stat_file.write(stats_instr_data_format.format(instr,
+                                                      self.manycore_stat_dict[instr],
+                                                      (100 * self.manycore_stat_dict[instr] / self.total_instr_cnt)))
+
+    stat_file.write(stats_instr_data_format.format("total",
+                                                   self.total_instr_cnt, 
+                                                   (100 * self.total_instr_cnt / self.total_instr_cnt)))
+    stat_file.write("{}\n".format(separating_line))
+    return
+
+
+  # Print instruction stats for each tile in a separate file 
+  # y,x are tile coordinates 
+  def print_per_tile_stats_instructions(self, y, x, stat_file):
```

Comment:
Well, I don't quite understand what you mean, the original point of this code was to print all stats into files, plus before printing we call generate functions that calculate all the stats. The reason for having this many different print functions is because each stats type (instruction, timing, miss, stall) has its own format so I had to design separate print functions for them. What I mean is even if I return a dictionary here, I'd still have to design different print functions to print the dictionary. So I'm not sure what returning a dictionary or string achieves.

Reply:
It means that someone else can reuse your code, and print it out as per their requirements.

Reply:
That's actually a good point! modified the class to return manycore stats, per_tile stats and timing stats dictionaries. printing occurs outside now. Thanks!